src/services/dataSourceFetcher.ts

```diff
@@ -12,58 +11,119 @@ export type DataSourceFetcherContentTypes = "json" | "csv" | "tsv";
 export class DataSourceFetcher extends AbstractDataFetcher {
     public constructor(src: string) {
         super(src);
-        this.src = src;
+        this._src = src;
```

Comment:
This was a hack to do with my `AbstractDataFetcher`
I had some grand plans with data fetchers so that pagination, data source fetching, etc... would all use the same base class then implement their own caching strategy

But I didn't end up implementing this functionality

Since `AbstractDataFetcher` is only used by this one class, and its implementation results in me defining two "sources of truth" (one on the DataSourceFetcher, and the AbstractDataFetcher), I've decided to remove the AbstractDataFetcher to make the code easier to read, and less prone to bugs.

Tl;DR - I've removed `_src`, because `AbstractDataFetcher` is no longer needed